Tests/LinAlg/test_LinAlg_inner_product.cpp

```diff
@@ -0,0 +1,76 @@
+#include "../test.hpp"
+
+NUMERIC_NSL_TEST_CASE("LinAlg: inner_product 1D", "[LinAlg,inner_product]"){
+    const NSL::size_t size = GENERATE(1,100,200,500,1000);
+    INFO(fmt::format("Size: {}", size));
+
+    // define 2 tensors with ones everywhere
+    NSL::Tensor<TestType> a(size); a = 1; 
+    NSL::Tensor<TestType> b(size); b = 2;
+
+    TestType res = NSL::LinAlg::inner_product(a,b);
+    REQUIRE( res == TestType(2*size) );
+    res = NSL::LinAlg::inner_product(a,a);
+    REQUIRE( res == TestType(size) );
+    res = NSL::LinAlg::inner_product(b,b);
+    REQUIRE( res == TestType(4*size) );
+}
+
+
+NUMERIC_NSL_TEST_CASE("LinAlg: inner_product 2D", "[LinAlg,inner_product]"){
+    const NSL::size_t size1 = GENERATE(1,100,200,500,1000);
+    const NSL::size_t size2 = GENERATE(1,100,200,500,1000);
+    INFO(fmt::format("Size: {}", size1));
+    INFO(fmt::format("Size: {}", size2));
+
+    // define 2 tensors with ones/twos everywhere such that we can 
+    // predetermine the results of the inner product
+    NSL::Tensor<TestType> a(size1,size2); a = 1; 
+    NSL::Tensor<TestType> b(size1,size2); b = 2;
+
+    TestType res = NSL::LinAlg::inner_product(a,b);
+    REQUIRE( res == TestType(2*size1*size2) );
+    res = NSL::LinAlg::inner_product(a,a);
+    REQUIRE( res == TestType(size1*size2) );
+    res = NSL::LinAlg::inner_product(b,b);
+    REQUIRE( res == TestType(4*size1*size2) );
+}
+
+
+FLOAT_NSL_TEST_CASE("LinAlg: inner_product random", "[LinAlg,inner_product]"){
+    const NSL::size_t size1 = GENERATE(1,100,200);
+    const NSL::size_t size2 = GENERATE(1,100,200);
+    INFO(fmt::format("Size: {}", size1));
+    INFO(fmt::format("Size: {}", size2));
+
+    // define 2 tensors randomly filled for checkup
+    NSL::Tensor<TestType> a(size1,size2); a.rand(); 
+    NSL::Tensor<TestType> b(size1,size2); b.rand();
```

Comment:
What does `.rand()` do for complex types?  Does it fill the imaginary parts?  If so then I think this test closes #144.

Reply:
Yes, that is exactly what it is doing. With this we can close #144 as well.